panic when MaxSessions is 1

If `MaxSessions` is set to 1, the following panic occurs:
```
panic: invalid/out of range capacity

goroutine 1 [running]:
github.com/thales-e-security/pool.NewResourcePool(0xc00629c030, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/cbroglie/.../vendor/github.com/thales-e-security/pool/resource_pool.go:87 +0x481
github.com/ThalesIgnite/crypto11.Configure(0xc0000caba0, 0x0, 0x0, 0x0)
	/Users/cbroglie/.../vendor/github.com/ThalesIgnite/crypto11/crypto11.go:327 +0x4bd
```
This is due to:
```go
// We will use one session to keep state alive, so the pool gets maxSessions - 1
instance.pool = pool.NewResourcePool(instance.resourcePoolFactoryFunc, maxSessions-1, maxSessions-1, 0, 0)
```

--- a/crypto11.go
+++ b/crypto11.go
@@ -235,6 +235,7 @@ type Config struct {
 	Pin string
 
 	// Maximum number of concurrent sessions to open. If zero, DefaultMaxSessions is used.
+	// Otherwise, the value specified must be at least 2.
 	MaxSessions int
 
 	// Maximum time to wait for a session from the sessions pool. Zero means wait indefinitely.
@@ -276,6 +277,9 @@ func Configure(config *Config) (*Context, error) {
 
 	if config.MaxSessions == 0 {
 		config.MaxSessions = DefaultMaxSessions
+	}
+	if config.MaxSessions == 1 {
+		return nil, errors.New("MaxSessions must be larger than 1")
 	}
 
 	instance := &Context{

--- a/crypto11_test.go
+++ b/crypto11_test.go
@@ -278,6 +278,15 @@ func TestNoLogin(t *testing.T) {
 	assert.Equal(t, pkcs11.Error(pkcs11.CKR_USER_NOT_LOGGED_IN), p11Err)
 }
 
+func TestInvalidMaxSessions(t *testing.T) {
+	cfg, err := getConfig("config")
+	require.NoError(t, err)
+
+	cfg.MaxSessions = 1
+	_, err = Configure(cfg)
+	require.Error(t, err)
+}
+
 // randomBytes returns 32 random bytes.
 func randomBytes() []byte {
 	result := make([]byte, 32)

